source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -17,16 +17,17 @@ namespace
     }
 }
 
-TypeDesc::TypeDesc(const string& name, unsigned char basetype, unsigned char semantic, int size, bool editable)
+TypeDesc::TypeDesc(const string& name, unsigned char basetype, unsigned char semantic, int size, bool editable, const std::unordered_map<char, int>& channelMapping)
```

Comment:
Small thing: For code cleanness, maybe add a "using" statement for std::unordered_map<char, int> ?

Reply:
+1, could be named ChannelMap or something similar.